api/renter_test.go

```diff
@@ -767,3 +645,377 @@ func TestRenterRelativePathErrorDownload(t *testing.T) {
+		t.Error("price changed from single to multi")
+	}
+}
+
+// TestRenterPricesHandlerPricey checks that the prices command returns
+// reasonable values given the settings of the hosts.
+func TestRenterPricesHandlerPricey(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	t.Parallel()
+	st, err := createServerTester("TestHostDBHostsAllHandlerPricey")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer st.server.Close()
+
+	// Announce the host and then get the calculated prices for when there is a
+	// single host.
+	var rpeSingle modules.RenterPriceEstimation
+	if err = st.announceHost(); err != nil {
+		t.Fatal(err)
+	}
+	if err = st.getAPI("/renter/prices", &rpeSingle); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create several more hosts all using the default settings.
+	stHost1, err := blankServerTester("TestRenterPricesHandlerPricey - Host 1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	stHost2, err := blankServerTester("TestRenterPricesHandlerPricey - Host 2")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var hg HostGET
+	err = st.getAPI("/host", &hg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost1.getAPI("/host", &hg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost2.getAPI("/host", &hg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Set host 5 to be cheaper than the rest by a substantial amount. This
+	// should result in a reduction for the price estimation.
+	vals := url.Values{}
+	vals.Set("mindownloadbandwidthprice", "100000000000000000000")
+	vals.Set("mincontractprice", "1000000000000000000000000000")
+	vals.Set("minstorageprice", "100000000000000000000")
+	vals.Set("minuploadbandwidthprice", "100000000000000000000")
+	err = stHost2.stdPostAPI("/host", vals)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Connect all the nodes and announce all of the hosts.
+	sts := []*serverTester{st, stHost1, stHost2}
+	err = fullyConnectNodes(sts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = fundAllNodes(sts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = announceAllHosts(sts)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Grab the price estimates for when there are a bunch of hosts with the
+	// same stats.
+	var rpeMulti modules.RenterPriceEstimation
+	if err = st.announceHost(); err != nil {
+		t.Fatal(err)
+	}
+	if err = st.getAPI("/renter/prices", &rpeMulti); err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that the aggregate is the same.
+	if !(rpeMulti.DownloadTerabyte.Cmp(rpeSingle.DownloadTerabyte) > 0) {
+		t.Error("price did not drop from single to multi")
+	}
+	if !(rpeMulti.FormContracts.Cmp(rpeSingle.FormContracts) > 0) {
+		t.Log(rpeMulti.FormContracts)
+		t.Log(rpeSingle.FormContracts)
+		t.Error("price did not drop from single to multi")
+	}
+	if !(rpeMulti.StorageTerabyteMonth.Cmp(rpeSingle.StorageTerabyteMonth) > 0) {
+		t.Error("price did not drop from single to multi")
+	}
+	if !(rpeMulti.UploadTerabyte.Cmp(rpeSingle.UploadTerabyte) > 0) {
+		t.Error("price did not drop from single to multi")
+	}
+}
```

Comment:
I felt kind of bad because these tests take like 8 seconds to run, but they caught bugs in I think 4 different packages, so I don't feel bad after all.